### PR TITLE
Add collections mapping

### PR DESCRIPTION
**Pull request by amitit**

### CODE REVIEW

File: gobserve.go [hunk 1]
```diff
@@ -26,14 +26,15 @@ type Subscription struct {
...
+	Handler        Handler
+	registry       *Registry
+	DB             *mgo.Database
+	watcherConfig  watcherConfig
+	handlerConfig  handlerConfig
+	registryConfig registryConfig
 }
 
-func NewGobserve(db *mgo.Database, collections []string, options ...GobserveOption) *Gobserve {
+func NewGobserve(db *mgo.Database, allowedKinds []string, options ...GobserveOption) *Gobserve {
```
Comment (sfriedel) [inline gobserve.go]: Having both allowedKinds and the collection mapping seems redundant to me. I'd prefer if we had a mandatory option for the kind mapping and everything that isn't present in there is not allowed by default.

Comment (sfriedel) [inline gobserve.go]: Actually I don't see it as much as kind mapping. It feels more like an essential configuration of exposed data types/subscription channels.

Comment (sfriedel) [inline gobserve.go]: My suggested naming would therefore be `resourceConfig` which should be a member of the `Gobserve` struct

File: gobserve.go [hunk 2]
```diff
@@ -8,8 +8,8 @@ import (
 )
 
 type Resource struct {
-	Collection string `json:"collection"`
-	Id         string `json:"resource_id"`
+	Kind string `json:"kind"`
```
Comment (sfriedel) [inline gobserve.go]: We should either have `kind` and `id` or `resource_kind` and `resource_id`. Right now it feels like a weird mix of both :)

**Comment by sfriedel** [comment]
@amitit handling looks fine. I'd like to improve the configuration part because there seems to be some redundancy left.

File: registry.go [hunk 3]
```diff
@@ -79,7 +81,12 @@ func (r *Registry) startWatcher(resource Resource) {
 	if _, ok := r.watchers[resource]; ok {
 		return
 	}
-	collection := r.gobserve.DB.C(resource.Collection)
+
+	collectionName := r.dataMapping.KindToCollection[resource.Kind]
+	if collectionName == "" {
+		log.Println("error " + resource.Kind + "not supported")
```
Comment (amitit) [inline registry.go]: @sfriedel we might need some error handling here, otherwise it's a silent fail

File: gobserve.go (hunk 2 above)
Comment (sfriedel) [inline gobserve.go]: @amitit this is still unsolved. Also `gofmt` is your friend ;)

Comment (amitit) [inline gobserve.go]: @sef I prefer `kind` and `id`

File: gobserve.go [hunk 4]
```diff
@@ -8,8 +8,8 @@ import (
...
+	Id   string `json:"id"`
```
Comment (amitit) [inline gobserve.go]: @sef done

**Comment by sfriedel** [comment]
@amitit still not quite what I had in mind. I'll try to elaborate on the intended parameterization:

```
type ResourceConfig {
	Collection string
}

type ResourceMap map[string]ResourceConfig

func main() {
	rmap := ResourceMap{
		"train": ResourceConfig{Collection: "trains"}
	}
	g := NewGobserve(db, rmap, ...)
}
```

After thinking more about this I'm currently even sympathetic to the following:

```
type ResourceKind string

type ResourceConfig {
	Collection *mgo.Collection
}

type ResourceMap map[string]ResourceConfig

func main() {
	rmap := ResourceMap{
		"train": ResourceConfig{Collection: db.C("trains")}
	}
	g := NewGobserve(rmap, ...)
}
```

If we did the latter we would also solve the multiple database problem.

@amitit what do you think?

**Comment by amitit** [comment]
@sfriedel I like the flexibility and the straight-forwardness of this solution. However to make this a real Config of a Resource we would better add a config attribute to `Resource`

```
type Resource struct {
	Kind   string `json:"kind"`
	Id     string `json:"id"`
	Config ResourceConfig
}
```

In this way a `Resource` is self contained and has all the attributes necessary for the polling.

This also may save us some parameters on function calls where both the `Resource` and its `Collection` are provided, such as `watcher.go:34 func NewWatcher`.

**Comment by sfriedel** [comment]
@amitit i wouldn't do this because Resource defines a single resource with an kind/id configuration. Probably `ResourceConfig` is a bad name because the configuration is for all resources with the same kind and i would welcome a better name for this.

**Comment by amitit** [comment]
@sfriedel please review

File: registry.go [hunk 5]
```diff
@@ -79,7 +81,12 @@ func (r *Registry) startWatcher(resource Resource) {
 	if _, ok := r.watchers[resource]; ok {
 		return
 	}
-	collection := r.gobserve.DB.C(resource.Collection)
+
+	resourceKindConfig := r.resourceKindMapping[resource.Kind]
+	if (resourceKindConfig == ResourceKindConfig{}) {
+		log.Println("error " + resource.Kind + "not supported")
```
Comment (sfriedel) [inline registry.go]: please prefer Printf wiht proper formatting to string concatenation

**Comment by sfriedel** [comment]
@amitit review done, just some very minor parts left

File: registry.go (hunk 5 above)
Comment (amitit) [inline registry.go]: done @sfriedel

**Comment by sfriedel** [comment]
@ajk LGTM, can be merged

**Pull request merged**